_posts/2016-12-05-haskell-package-checklist.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Haskell package checklist
+---
+
+- Host on GitHub
+  - Easy for other developers to contribute.
+  - Integrates well with other services.
+  - More popular than other hosts.
+- Build with Stack
+  - Easiest way to manage dependencies.
+  - Ensures that your package will remain buildable.
+  - Makes difficult workflows easy, like `--profile` and `--coverage`.
+- Use `hpack`
+  - Uses YAML instead of custom package format.
+  - Avoids unnecessary boilerplate.
+  - Integrated directly into Stack.
+- Name package with `kebab-case`
+  - Keep everything lowercase to avoid confusion.
+  - Use hyphens to separate words.
+  - Keep it short and memorable.
+- Use Semantic Versioning
+  - SemVer is used by many other languages.
+  - It matches how developers generally think about versions.
+  - Hackage recommends the PVP, which adds ambiguity.
```

Comment:
By "ambiguity", I mean that the PVP has two major version numbers. It's not clear when you should increment one versus the other. 

Also I want to add that the PVP does not treat major version 0 specially like SemVer does. That means rock-solid core libraries like `containers` and `bytestring` end up with 0 as their major versions. It's not necessarily a problem, but it doesn't inspire confidence.